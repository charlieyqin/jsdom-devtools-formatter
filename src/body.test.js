const body = require("./body");

describe("body", () => {
  it("returns expected JsonML for 1st-level elements", () => {
    const element = document.createElement("div");
    element.innerHTML = `
      <h1 id="title">A title</h1>
      <article class="example">
        <p>Lorem ipsum</p>
        <p>and so forth</p>
      </article>
      <div class="">
        <!-- a comment -->
      </div>
      <footer>
        Foo bar
      </footer>
    `;
    const actual = body(element);
    expect(actual).toMatchSnapshot();
  });
});
