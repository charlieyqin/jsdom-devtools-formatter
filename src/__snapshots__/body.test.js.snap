// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`body returns expected JsonML for 1st-level elements 1`] = `
Array [
  "ol",
  Object {
    "style": "
    list-style-type: none;
    margin: 2px 0 0 28px;
    padding: 0;
  ",
  },
  Array [
    "li",
    Object {
      "style": "
    margin-bottom: 2px;
  ",
    },
    "",
  ],
  Array [
    "li",
    Object {
      "style": "
    margin-bottom: 2px;
  ",
    },
    Array [
      "object",
      Object {
        "object": <h1
          id="title"
        >
          A title
        </h1>,
      },
    ],
  ],
  Array [
    "li",
    Object {
      "style": "
    margin-bottom: 2px;
  ",
    },
    "",
  ],
  Array [
    "li",
    Object {
      "style": "
    margin-bottom: 2px;
  ",
    },
    Array [
      "object",
      Object {
        "object": <article
          class="example"
        >
          
        
          <p>
            Lorem ipsum
          </p>
          
        
          <p>
            and so forth
          </p>
          
      
        </article>,
      },
    ],
  ],
  Array [
    "li",
    Object {
      "style": "
    margin-bottom: 2px;
  ",
    },
    "",
  ],
  Array [
    "li",
    Object {
      "style": "
    margin-bottom: 2px;
  ",
    },
    Array [
      "object",
      Object {
        "object": <div
          class=""
        >
          
        
          <!-- a comment -->
          
      
        </div>,
      },
    ],
  ],
  Array [
    "li",
    Object {
      "style": "
    margin-bottom: 2px;
  ",
    },
    "",
  ],
  Array [
    "li",
    Object {
      "style": "
    margin-bottom: 2px;
  ",
    },
    Array [
      "object",
      Object {
        "object": <footer>
          
        Foo bar
      
        </footer>,
      },
    ],
  ],
  Array [
    "li",
    Object {
      "style": "
    margin-bottom: 2px;
  ",
    },
    "",
  ],
]
`;
